src/v/redpanda/application.h

```diff
@@ -315,6 +315,48 @@ class application {
         }
     }
 
+    // Shutdown the service with custom method.
+    // The method should be invoked in the ss::thread context.
+    template<class StopFunc>
+    void shutdown_with_watchdog(
+      ss::sstring name, ss::sstring method_name, StopFunc&& stop_func) {
+        // This watchdog is triggered after short period of time (30s). It
+        // adds message to the log that service is taking a long time to
+        // shutdown on INFO level.
+        ssx::watchdog short_wd(
+          short_shutdown_warning_timeout, [this, name, method_name] {
+              vlog(
+                _log.info,
+                "Service {} is taking more than {} seconds to shutdown with "
+                "{}.",
+                name,
+                std::chrono::duration_cast<std::chrono::seconds>(
+                  short_shutdown_warning_timeout)
+                  .count(),
+                method_name);
+          });
+        // This watchdog is triggered after long period of time. This indicates
+        // a bug (most likely).
+        ssx::watchdog long_wd(
+          long_shutdown_warning_timeout, [this, name, method_name] {
+              vlog(
+                _log.error,
+                "Service {} is taking more than {} seconds to shutdown with "
+                "{}!",
+                name,
+                std::chrono::duration_cast<std::chrono::seconds>(
+                  long_shutdown_warning_timeout)
+                  .count(),
+                method_name);
+          });
+
+        vlog(_log.info, "Shutting down {} with {} method", name, method_name);
```

Comment:
nit: is info intentional in this line?

Reply:
yes, i want it to appear everywhere as this is low volume and very helpful